CMakeLists.txt

```diff
@@ -6,8 +6,13 @@ list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 
 # Prefer our bundled versions of dependencies.
 set(DEPS_PREFIX "${CMAKE_CURRENT_SOURCE_DIR}/.deps/usr" CACHE PATH "Path prefix for finding dependencies")
-list(INSERT CMAKE_PREFIX_PATH 0 ${DEPS_PREFIX})
-set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:${DEPS_PREFIX}/lib/pkgconfig")
+if(CMAKE_CROSSCOMPILING)
+  list(INSERT CMAKE_FIND_ROOT_PATH 0 ${DEPS_PREFIX})
+  list(INSERT CMAKE_PREFIX_PATH 0 ${DEPS_PREFIX}/../host/bin)
```

Comment:
/host/ comes from third-party/cmake/Build* recipes. e.g. when cross compiling we need to build luajit and libuv both for the host and target system. Libs for the target system are installed in .deps/usr/ and libs for the host are placed in .deps/host. (here is an [example](https://github.com/neovim/neovim/blob/master/third-party/cmake/BuildLibuv.cmake#L48))

> Also note that this PR is currently breaking travis.

Weird, but it is only breaking Travis in the gcc-5-m32 job. Ahh maybe the CMAKE_CROSSCOMPILING is evaluating as true for that case (because we are cross compiling from a 64 bit host to 32 bit target). Does changing `if(CMAKE_CROSSCOMPILING)` to `if(MINGW AND CMAKE_CROSSCOMPILING)` help? Still does not explain why the error is happening though.

> think we need to set -m32 in the linker flags in cmake/i386-linux-gnu.toolchain.cmake

Its already there.